packages/ai-bot/lib/send-response.ts

```diff
@@ -81,38 +87,47 @@ export class Responder {
 
   async onMessage(msg: {
     role: string;
-    tool_calls?: { function: { name: string; arguments: string } }[];
+    tool_calls?: { function: FunctionToolCall }[];
   }) {
     if (msg.role === 'assistant') {
-      for (const toolCall of msg.tool_calls || []) {
-        const functionCall = toolCall.function;
-        log.debug('[Room Timeline] Function call', toolCall);
-        let args;
-        try {
-          args = JSON.parse(functionCall.arguments);
-        } catch (error) {
-          Sentry.captureException(error);
-          this.initialMessageReplaced = true;
-          let errorPromise = sendError(
-            this.client,
-            this.roomId,
-            error,
-            this.initialMessageReplaced ? undefined : this.initialMessageId,
-          );
-          this.messagePromises.push(errorPromise);
-          await errorPromise;
-        }
-        if (functionCall.name === 'patchCard') {
+      await this.handleFunctionToolCalls(msg);
+    }
+  }
+
+  async handleFunctionToolCalls(msg: {
+    role: string;
+    tool_calls?: { function: FunctionToolCall }[];
+  }) {
+    for (const toolCall of msg.tool_calls || []) {
+      const functionCall = toolCall.function;
+      log.debug('[Room Timeline] Function call', toolCall);
+      try {
+        functionCall.arguments = JSON.parse(functionCall.arguments);
```

Comment:
if'm confused by the types here--how do we know this type is a string? in the code above you treated this properly as an object and now it's a string. what is going on here? this is another reason why `any` is a bad idea. ther eis something happening here that could be illustrated using the better typing, but it's confusing what is going on.

Reply:
Copied this from 



![Screenshot 2024-06-12 at 01 06 03](https://github.com/cardstack/boxel/assets/8165111/b682344a-954a-4d17-9f44-1e8cc7577c38)
![Screenshot 2024-06-12 at 01 06 03](https://github.com/cardstack/boxel/assets/8165111/b682344a-954a-4d17-9f44-1e8cc7577c38)
![Screenshot 2024-06-12 at 01 06 03](https://github.com/cardstack/boxel/assets/8165111/b682344a-954a-4d17-9f44-1e8cc7577c38)
somewheer. but this is the reason

Reply:
it would be nice if we have a way to go between these 2 types: open-ai-serialized and open-ai-deserialized